criu/kerndat.c

```diff
@@ -1242,8 +1308,15 @@ int kerndat_init(void)
 	int ret;
 
 	ret = kerndat_try_load_cache();
-	if (ret <= 0)
+	if (ret < 0)
 		return ret;
+
+	/* Try to collect missing hugetlb dev in kerndat cache */
+	if (ret == 0) {
+		ret = kerndat_get_hugetlb_dev();
+		return ret;
+	}
```

Comment:
Please add a comment why we have to re-detect hugetlb devices after successful load of the cache. Don't we want to update the cache here as well?

Also, you can simply this with
```
	if (ret == 0)
		return kerndat_get_hugetlb_dev();
```

Reply:
I forget to update the cache here.
`kerndat_get_hugetlb_dev()` only tries to update the device number that at the previous time we check, we see that hugetlb size is supported but we cannot mmaped to get that device number. As the hugetlb page can be adjusted between run, the hugetlb size that fails before may be available now.
I will adjust `kerndat_get_hugetlb_dev()` so that it will return -1 if there is an error, 0 if it is successful but find nothing new, 1 if it is successful and we find some new hugetlb device number. We will only update the cache in case that function return 1